### PR TITLE
Initial inventory generation playbooks

### PR DESCRIPTION
**Pull request by tylerauerbeck**
This creates the initial inventory generation playbooks. I'm not attached to any of the structure here, so if this doesn't work for anybody else let me know and we can move things around.

### CODE REVIEW

File: inventory-generation/identity-management/site.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  - name: "Fail If Working Directory Is Not Provided"
+    fail:
+      msg: "directory var needs to be provided in order to ensure a safe working directory"
+    when:
+      - directory is undefined or (directory | trim) == ""
+
+  - name: "Read Engagement Data"
+    set_fact:
+      json_data: "{{ (lookup('file', directory + '/iac/engagement.json') | from_json).engagement_users | to_json }}"
```
Comment (pabrahamsson) [inline inventory-generation/identity-management/site.yml]: I believe the correct path is `directory + '/engagement.json'`. See [here](/rht-labs/lodestar-automation/blob/master/translate-engagement-data/site.yml#L8) for example.

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
Looks good @tylerauerbeck but I think the path on the filesystem might need to change. See inline comment for details.

File: inventory-generation/identity-management/site.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      - "ocp-ldap-sa-credentials.json"
+
+  - name: "Add users to inventory"
+    set_fact:
+      users: "{{ (users | default([])) +  [ { 'first_name': (item.first_name | trim), 'last_name': (item.last_name | trim), 'email': (item.email | trim) , 'user_name': (item.first_name | trim | lower)[0] + (item.last_name | trim | lower) } ] }}"
+    loop: "{{ engagement_users }}"
+
+  - name: "Add LDAP Service Account"
+    set_fact:
+      users: "{{ (users | default([])) + [ { 'first_name': 'LDAP', 'last_name': 'SA', 'email': 'ldap@localhost', 'user_name': ocp_ldap_sa_username, 'generate_password': false, 'password': ocp_ldap_sa_password  } ] }}"
```
Comment (pabrahamsson) [inline inventory-generation/identity-management/site.yml]: The IdM doesn't like `ldap@localhost`
```
response user_add: invalid 'email': invalid e-mail format: ldap@localhost
```

File: inventory-generation/identity-management/site.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  - name: "Get Unique Groups"
+    set_fact:
+      unique_groups: "{{ engagement_users | json_query('[].role') | unique }}"
+
+  - name: "Set Group Membership"
+    set_fact:
+      usrgrp: "{{ (usrgrp | default([])) + [ {'name': item, 'members': (engagement_users |  json_query(query)) } ] }}"
+    vars:
+      query: "[?role =='{{ item }}'].email"
```
Comment (pabrahamsson) [inline inventory-generation/identity-management/site.yml]: This tries to use the email address as the key for group membership but should be the `user_name` per [the docs](https://github.com/redhat-cop/infra-ansible/blob/master/roles/identity-management/README.md).

File: inventory-generation/identity-management/site.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    loop: "{{ unique_groups }}" 
+
+  - name: Add LDAP Groups Grouping
+    set_fact:
+      usrgrp: "{{ (usrgrp | default([])) + [ {'name': 'ldap-users', 'childgroups': unique_groups } ] }}"
```
Comment (pabrahamsson) [inline inventory-generation/identity-management/site.yml]: `'s/ldap-users/ldap-members/`

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
I noticed a few things in the generated resourceclaim while testing, comments inline.

File: inventory-generation/identity-management/site.yml (hunk 1 above)
Comment (tylerauerbeck) [reply inventory-generation/identity-management/site.yml]: Fixed

File: inventory-generation/identity-management/site.yml (hunk 2 above)
Comment (tylerauerbeck) [reply inventory-generation/identity-management/site.yml]: Yeah, real email address formats are useful 👍 . Fixed.

File: inventory-generation/identity-management/site.yml (hunk 4 above)
Comment (tylerauerbeck) [reply inventory-generation/identity-management/site.yml]: Fixed.

**Review by jacobsee — APPROVED** [review]

**Pull request merged**